frontend/test/pytest/test_jax_integration.py

```diff
@@ -307,6 +307,32 @@ def cost_fn(x, y):
         assert jnp.allclose(result1, 0.0)
         assert jnp.allclose(result2, 0.0)
 
+    @pytest.mark.parametrize("shape", ([2, 3], [3, 2], [1, 6]))
+    def test_multiD_calls(self, backend, shape):
+        """Test a jax.grad in combination with qjit on non-1D input parameters."""
+
+        def mock_circuit(p1, p2):
+            return jnp.reshape(p1, shape) + 2 * jnp.reshape(p2, shape)
+
+        def circuit(p1, p2):
+            qml.RY(p1[0, 1], wires=0)
+            qml.RZ(p2[1, 0], wires=0)
+            return jnp.reshape(p1, shape) + 2 * jnp.reshape(p2, shape)
+
+        C_circuit = qjit(qml.qnode(qml.device(backend, wires=1))(circuit))
+        PL_circuit = jax.jit(mock_circuit)
+
+        def cost_fn(p1, p2, c):
+            m1 = c(p1, p2)
+            m2 = c(p1, p2)
+            return m1 + m2
+
+        p1 = jnp.array([[0.1, 0.3, 0.5], [0.1, 0.2, 0.8]])
+        p2 = jnp.array([[0.3, 0.5], [0.2, 0.8], [0.2, 0.8]])
+        result = jax.jacobian(partial(cost_fn, c=C_circuit))(p1, p2)
+        reference = jax.jacobian(partial(cost_fn, c=PL_circuit))(p1, p2)
+        assert jnp.allclose(result, reference, rtol=1e-6, atol=1e-6)
```

Comment:
A couple things are standing out to me about this test:

- `mock_circuit` and `circuit` and functionality the same, so I don't see what the point of circuit is
- I don't think we need to involve `jax.jit` for the reference computation, the simpler the test the better
- the function takes two arguments but is only differentiated against the first one, is that intended?
- asserting the shape (first) in addition to the values might be appropriate here, especially since allclose can perform broadcasting
- the use of `partial` is not needed since the differentiable arguments are already controlled via `argnums` which does not include the third argument

Reply:
Thanks @grwlf! Yeah that is a default we copied from JAX. I have no particular preference here, but staying close to the JAX API unless there is a good reason not to is probably the best experience for users.